### ISSUE TITLE
Feature debug_asset_server causes freeze/infinite loop on startup

### ISSUE THREAD

**Issue by JMS55** [bevyengine/bevy#7563]
Bevy version 978f7cd8bf9423961f663725cba18fb3ed75b46d.

**Comment by hymm** [comment]
Something is wrong with nesting multithreaded schedules. I was able to workaround the issue with
```rust
debug_asset_app
    .edit_schedule(CoreSchedule::Main, |schedule| {
        schedule.set_executor_kind(ExecutorKind::SingleThreaded);
    })
    .edit_schedule(CoreSchedule::Startup, |schedule| {
        schedule.set_executor_kind(ExecutorKind::SingleThreaded);
    });
```

**Comment by james7132** [comment]
This is fixed as of #7825.
